domain-services/fiscal-event-service/src/main/java/org/egov/validator/FiscalEventValidator.java

```diff
@@ -86,130 +87,87 @@ public void validateFiscalEventPushPost(FiscalEventRequest fiscalEventRequest) {
-                if (amountDetails == null || amountDetails.isEmpty()) {
-                    errorMap.put("AMOUNT_DETAILS", "Amount details are missing in the request");
-                    return;
-                }
-                allAmountDetails.addAll(amountDetails);
-            }
-
-            Set<String> coaCodes = new HashSet<>();
-            for (Amount amount : allAmountDetails) {
-                if (amount.getAmount() == null /*|| amount.getAmount().compareTo(BigDecimal.ZERO)==0*/)
-                    errorMap.put("AMOUNT", "Amount is missing in request for coaCode : " + amount.getCoaCode());
-                if (StringUtils.isNotBlank(amount.getCoaCode()))
-                    coaCodes.add(amount.getCoaCode());
-                else {
-                    errorMap.put("COA_CODE", "COA Code is missing in the request data");
-                    throw new CustomException(errorMap);
-                }
+        // Set<String> coaCodes = new HashSet<>();
+        for (Amount amount : amountDetails) {
+            if (amount.getAmount() == null /* || amount.getAmount().compareTo(BigDecimal.ZERO)==0 */)
+                errorMap.put("AMOUNT", "Amount is missing in request for coaCode : " + amount.getCoaCode());
+            if (StringUtils.isBlank(amount.getCoaCode())) {
+                errorMap.put("COA_CODE", "COA Code is missing in the request data");
+                throw new CustomException(errorMap);
             }
+        }
 
-            JsonNode jsonNode = coaUtil.fetchCoaDetailsByCoaCodes(requestHeader, coaCodes, tenantId);
-
-            List<String> errorCoaCodes = new ArrayList<>();
-            for (String coaCode : coaCodes) {
-                boolean isPresent = false;
+    }
 
-                if (jsonNode != null) {
-                    for (JsonNode chartOfAccountJN : jsonNode) {
-                        String coaCd = chartOfAccountJN.get("coaCode").asText();
-                        if (StringUtils.isNotBlank(coaCd) && coaCode.equals(coaCd.trim())) {
-                            isPresent = true;
+    private void validateParentEventId(Set<String> parentEventIds, String tenantId, Map<String, String> errorMap) {
+
+        if (!parentEventIds.isEmpty() && StringUtils.isNotBlank(tenantId)) {
+            Criteria criteria = new Criteria();
+            List<String> ids = new ArrayList<>(parentEventIds);
+            criteria.setTenantId(tenantId);
+            criteria.setIds(ids);
+
+            List<FiscalEvent> fiscalEventList = eventRepository.searchFiscalEvent(criteria);
+            if (fiscalEventList == null || fiscalEventList.isEmpty()) {
+                errorMap.put(MasterDataConstants.PARENT_EVENT_ID,
+                        "Linked event id(s) : " + parentEventIds + " doesn't exist in the system.");
+            } else if (!fiscalEventList.isEmpty() && (fiscalEventList.size() != parentEventIds.size())) {
+                /*
+                 * List<FiscalEvent> dbFiscalEvents = mapperUtil
```

Comment:
If there isn't any important reason to keep any of this code, then remove it instead of commenting it.

Reply:
minor change, to be taken up as part of code cleanup.